grails-app/controllers/org/broadinstitute/orsp/api/DataUseLetterController.groovy

```diff
@@ -55,4 +63,65 @@ class DataUseLetterController extends AuthenticatedController {
         }
     }
 
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
+    def createPdf() {
+        String templateName = "DataUseLetterPDF.pdf"
+        String uid = request.JSON[DataUseLetterFields.UID.abbreviation]
+        DataUseLetter dul = DataUseLetter.findByUid(uid)
+        if (dul != null) {
+            ByteArrayOutputStream output = new ByteArrayOutputStream()
+            PDDocument dulDOC = new PDDocument()
+            try {
+                ClassLoader classLoader = getClass().getClassLoader()
+                InputStream is = classLoader.getResourceAsStream(templateName)
+                dulDOC = PDDocument.load(is)
+                is.close()
+                new DulPdfParser().fillDulForm(dul, dulDOC.getDocumentCatalog().getAcroForm())
+                dulDOC.save(output)
+                String fileName = JSON.parse(dul.dulInfo)[DataUseLetterFields.PROTOCOL_TITLE.abbreviation].replaceAll(' / ','_').concat("_DataUseLetter.pdf")
+                uploadDataUseLetter(dul,new ByteArrayInputStream(output.toByteArray()), fileName)
+
+                dul.setSubmitted(true)
+                dul.setSubmitDate(new Date())
+                dul.save(flush:true)
+                dulDOC.close()
+                output.close()
+                response.status = 200
+            } catch (Exception e) {
+                response.status = 500
```

Comment:
I may have missed some earlier examples of this, but we should probably be logging these errors

Reply:
Got it!, I added a `log.error` here.